Simplenote/SPAppDelegate.m

```diff
@@ -295,6 +295,25 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     return YES;
 }
 
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray * _Nullable))restorationHandler {
+    
+    if (userActivity.userInfo != nil) {
+        if (userActivity.userInfo[CSSearchableItemActivityIdentifier] != nil) {
+            NSString *uniqueIdentifier = userActivity.userInfo[CSSearchableItemActivityIdentifier];
+            SPBucket *noteBucket = [_simperium bucketForName:@"Note"];
+            Note *note = [noteBucket objectForKey:uniqueIdentifier];
+            double delayInSeconds = 0.05;
+            dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));
+            dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
+                [_noteListViewController openNote:note fromIndexPath:nil animated:NO];
+                [self showPasscodeLockIfNecessary];
```

Comment:
I took this chunk from `- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url`. I'll extract it to a separate method, and reuse in those 2 scenarios. Seems to me that it's necessary in a case when the user leaves the app on Settings, then comes back to the app via Spotlight or URL. At this point the view stack needs to be cleaned up first, then we can show a new note (via URL) or a note from Spotlight Results.